### ISSUE TITLE
Can't import ECR repository resources

### ISSUE THREAD

**Issue by SydOps** [hashicorp/terraform#14604]
Hi there,

### Terraform Version
0.9.3

### Affected `Resource(s)`
- aws_ecr_repository

### Terraform Configuration Files

```hcl
variable "repository" {
  type    = "list"
  default = ["docker-1", "docker-2", "docker-3"]
}

resource "aws_ecr_repository" "repository" {
  count = "${length(var.repository)}"
  name  = "${element(var.repository, count.index)}"
}
```

### Panic Output
I successfully import the first repository resource `docker-1`, but can't import the rest.

>Error importing: 1 error(s) occurred:

>* aws_ecr_repository.repository (import id: docker-2): Can't import aws_ecr_repository.repository, would collide with an existing resource.

>Please remove or rename this resource before continuing.

### Expected Behavior
import all ECR repositories. 

### Actual Behavior
I successfully import the first repository resource `docker-1`, but can't import the rest.

>* aws_ecr_repository.repository (import id: docker-2): Can't import aws_ecr_repository.repository, would collide with an existing resource.

### Steps to Reproduce
1. Manually create ECR repo `docker-1, docker-2, docker-3`
2. run `terraform import aws_ecr_repository.repository docker-1`, get updated `terraform.tfstate` with the resource repository `docker-1`
3.  run `terraform import aws_ecr_repository.repository docker-2` or other repositories, get the error I list in  `Actual Behavior`

**Comment by SydOps** [comment]
I run the test `terraform apply` with new repositories to be created, and compare the differences of `terraform.tfstate` with `terraform import`

1. Delete the repositories (docker-1, docker-2, docker-3)
2. run `terraform apply`, three ECR repositories are created. 
3. related resource tfstate content: 

            "resources": {
                "aws_ecr_repository.repository.0": {
                    "type": "aws_ecr_repository",
                    "depends_on": [],
                    "primary": {
                        "id": "docker-1",
                        "attributes": {
                            "arn": "arn:aws:ecr:xxxx:xxxx:repository/docker-1",
                            "id": "docker-1",
                            "name": "docker-1",
                            "registry_id": "xxxx",
                            "repository_url": "xxxx.dkr.ecr.xxxx.amazonaws.com/docker-1"
                        },
                        "meta": {},
                        "tainted": false
                    },
                    "deposed": [],
                    "provider": ""
                },

So the resource name created by `terraform apply` are `aws_ecr_repository.repository.[0|1|2]`

4. rename terraform.tfstate to terraform.tfstate.bak

`    mv terraform.tfstate terraform.tfstate.bak`

5. Import the resources 

```
$ terraform import aws_ecr_repository.repository docker-1
aws_ecr_repository.repository: Importing from ID "docker-1"...
aws_ecr_repository.repository: Import complete!
  Imported aws_ecr_repository (ID: docker-1)
aws_ecr_repository.repository: Refreshing state... (ID: docker-1)

Import success! The resources imported are shown above. These are
now in your Terraform state. Import does not currently generate
configuration, so you must do this next. If you do not create configuration
for the above resources, then the next `terraform plan` will mark
them for destruction.

$ terraform import aws_ecr_repository.repository docker-2
aws_ecr_repository.repository: Importing from ID "docker-2"...
aws_ecr_repository.repository: Import complete!
  Imported aws_ecr_repository (ID: docker-2)
Error importing: 1 error(s) occurred:

* aws_ecr_repository.repository (import id: docker-2): Can't import aws_ecr_repository.repository, would collide with an existing resource.

Please remove or rename this resource before continuing.

$ terraform import aws_ecr_repository.repository docker-1
aws_ecr_repository.repository: Importing from ID "docker-1"...
aws_ecr_repository.repository: Import complete!
  Imported aws_ecr_repository (ID: docker-1)
Error importing: 1 error(s) occurred:

* aws_ecr_repository.repository (import id: docker-1): Can't import aws_ecr_repository.repository, would collide with an existing resource.

Please remove or rename this resource before continuing.
```

6. check the new `terraform.tfstate`, the resource name is `aws_ecr_repository.repository`. No number after it. 

Maybe that's the problem?

**Comment by SydOps** [comment]
I went through the close ticket and PRs and found the problem has been reported and fixed

Problem was reported by #7691

Fixed by #8335

But why I can't import the resource.

**Comment by SydOps** [comment]
Successfully import the resource.

    terraform import aws_ecr_repository.repository[0] docker-1
    terraform import aws_ecr_repository.repository[1] docker-2
    terraform import aws_ecr_repository.repository[2] docker-3
